### ISSUE TITLE
Eliptic curve certificates

### ISSUE THREAD

**Issue by jakobbotsch** [thery/coqprime#18]
According to the readme and old documentation, coqprime should support checking eliptic curve certificates generated by Primo. However I cannot seem to get `o2v` working with certificates generated by Primo. Is there a step-by-step guide on this somewhere, or what is the status of this support?

**Comment by thery** [comment]
It still works. Here how to do.
First you need to put ```primo``` is a special configuration. This is done by adding the two lines
```
[Setup+]
ECO=1
BHB=1
```
for example my ```primo.ini``` looks like

```[Setup]
CertificationReports=1
CheckSignaturesOnly=1
MaxConcurrentTasks=8
SieveDivisorSize=26
SieveNumberSize=5000

[Setup+]
ECO=1
BHB=1
```

now if I have a file ```primo-NNN.out```

the command
````o2v -split --split -n my_thm -o my_thm_file primo-NNN.out````
will generate a bunch of files ```my_thm_file_NN.v```

Each file  ```my_thm_file_NN.v``` proves an implication of the form ```prime N1 -> prime N2``` where ```N1``` is smaller than ```N2```. The file ```my_thm_file.v``` glues everything together and prove the theorem ```my_thm``` that asserts the primality of the number attached to ```primo-NNN.out```.

Hope this helps

**Comment by jakobbotsch** [comment]
Thanks for the response. Unfortunately I still cannot get it to work. Here is a sample certificate: https://gist.github.com/jakobbotsch/b4d8d4bf7ec8e9c72f6a1581a7994fc3

These regexes do not seem to match anything:
https://github.com/thery/coqprime/blob/f38fff80b6d7b4bded5a3e91d10d9ebcb4ca1588/gencertif/parser.ml#L10-L16
The `$` is on the wrong side of the `=` compared to in the certificate. I am not sure if perhaps the Primo certificate format has changed?

**Comment by thery** [comment]
you are right. Unfortunately the fix is not so direct because of negative numbers.
I am using 4.1.1 and the certificate is ok

[primo-B409E03229D51-001.txt](https://github.com/thery/coqprime/files/4059417/primo-B409E03229D51-001.txt)

For the time being if you can get 4.1.1 you can send me the .in I will send you back the .out

Sorry about that

**Comment by jakobbotsch** [comment]
Thank you, that seems to generate proper Coq proofs, although it does not seem quick enough to check it.
Does Coqprime use primitive integers yet? I can see that it requires bignums as a package, but I also notice that there are some uses of normal Z. Just wondering if there is any low hanging fruit left to possibly allow Coqprime to verify a prime of this size (roughly 670 digits).

**Comment by thery** [comment]
@jakobbotsch  I've just added support of integer in the last version (it has already been published on opam). WIth it I have been capable to check the primality of a 1200 decimal digit numbers in 9h (in fact 32m since I've been using the -split to do parallelel computation).

**Comment by jakobbotsch** [comment]
Perfect, very nice! Thanks for the help.

**Comment by thery** [comment]
@jakobbotsch on my 4-core laptop your prime is proved in 15m (1h real time)

**Comment by thery** [comment]
@jakobbotsch I have just pushed a new version of ```o2v``` that should be compatible with the last version 
of ```primo``` (there is some documentation about how to configurate primo in the main README of coprime).
If you could give it a try and see if you manage to get everything working, it would be very much appreciated.
Thanks

**Comment by jakobbotsch** [comment]
Yes it works from start to finish now, thank you for the quick fixes and support.

I can see you updated the readme already, but it would be useful to point out how to use the library, similarly to what was in the old readme at http://coqprime.gforge.inria.fr/ but updated for the current version. I did not even realize that coqprime also formalizes elliptic curve certificates before I stumbled upon the old page by chance.
